### DISCUSSION TITLE
How to make alpineJs not open all modal from list of items using "dispatch"

### DISCUSSION THREAD

**Discussion by ericknyoto** [livewire/livewire#3552] [1. Help]
Hi,
My codes looks like this 

Basically, I want to show edit user modal in a list of items, but to toggle the modal, I use Alpine js dispatch. How I avoid to open all edit user modal when I clicked on specific user.

For now, if I clicked on the edit button, the modal will show as many as users are on the list, if the lists have 10 users, then the modal will show 10 times.

`index.blade.php` -> this is a livewire view, So i have another livewire view inside a livewire view (`<livewire:edit-user />`)

```
@foreach ($users as $user)
<div>{{ $user-name}} </div>
<button @click=" isOpen = false $dispatch('show-edit-modal-with-data')>Edit</button>
<livewire:edit-user />
@endforeach
```

livewire\edit-user.blade.php
`<div @show-edit-modal-data="isOpen = true>Inside this div are data from livewire EditUser class</div>`

Did you think, my approach is imposible?
I come to this because, I don't want to send ajax request to only show modal.
I also notice that usually people told to use @entangle to toggle modal.
Any help, appreciated. Thanks!

**Comment by joshhanley** [comment]
@ericknyoto yeah you would need to scope your event with some form of ID.

Or as you said, you could make use of `@entangle`, but would probably trigger an ajax request.

I think scoping would work, just add the user ID to the event payload, and check for it in the component.

```js
$dispatch('show-edit-modal-with-data', { userId: {{ $user->id }} })
```

Then inside you model, check, 

```blade
<div @show-edit-modal-data="if($event.detail.userId == {{ $user->id }}) { isOpen = true; }">Inside this div are data from livewire EditUser class</div>
```

Something like that. Note, I've just written it off the top of my head so haven't tested it, but should give you a good idea.

Hope this helps!
